Auto convert using Grafana 11 dashboards

dougszumski opened the pull request:
- Load dashboards into Grafana 11 preview
- View dashboard and export
- Check in changes
  - Don't allow hard coded Prometheus datasources

Needs reviewing / testing.

dougszumski committed:
Upgrade some dashboards using Grafana 11 autoconversion

Load a group of dashboards into Grafana 11, check that they
function and check in the changes.

markgoddard commented:
"pluginVersion": "11.0.0-preview", looks a bit weird

dougszumski commented:
> "pluginVersion": "11.0.0-preview", looks a bit weird

Yeah, 11.0.0 wasn't out then, but it is now (the main advantage for this is that it provides auto-conversion). Dawud did test that they would load back into 10.x.

We can remove `-preview` now.

dougszumski commented:
Please can you test this @technowhizz ?

I have switched to the non-preview plugin version.

technowhizz commented on etc/kayobe/kolla/config/grafana/dashboards/openstack/node_exporter_full.json:
The System Processes part of this dashboard is collapsed and so it appears that they haven't been converted

technowhizz replied:
I fixed this in #1069 but it is still the 11.0-preview one

dougszumski replied:
Thanks for that- you can use `sed` to strip off the -preview. I found that worked fine. I will remove node exporter from this, in favour of your PR

dougszumski replied:
I've removed this in favour of your PR

technowhizz commented on etc/kayobe/kolla/config/grafana/dashboards/openstack/rabbitmq.json:
The Queues, Channels and Connections sections have blank panels. The PQL is returning valid data. There seems to be an issue with the panel itself

dougszumski replied:
Thanks, will investigate

dougszumski replied:
It took me ages to find, but the trick is to remove this line:

![image](https://github.com/stackhpc/stackhpc-kayobe-config/assets/1140727/a650d8cf-995d-40e6-a822-e4b445d99fc4)

Then it works as expected!

dougszumski committed:
Fix decimals field which breaks plots

These plots show integer quanties, so set decimals to 0. It isn't
clear where the -1 setting came from, but it breaks the plot entirely
on Grafana 11.

technowhizz reviewed (approved):
LGTM

dougszumski committed:
Merge branch 'stackhpc/2023.1' into bugfix/INFRA-629-2
